tee-worker/omni-executor/rpc-server/src/methods/omni/request_loan.rs

```diff
@@ -0,0 +1,166 @@
+use super::common::handle_omni_native_task;
+use crate::detailed_error::DetailedError;
+use crate::error_code::{AUTH_VERIFICATION_FAILED_CODE, PARSE_ERROR_CODE};
+use crate::methods::omni::common::check_auth;
+use crate::methods::omni::PumpxRpcError;
+use crate::server::RpcContext;
+use crate::validation_helpers::{
+	validate_chain_id, validate_omni_account_hex, validate_omni_account_length,
+	validate_wallet_index,
+};
+use executor_core::intent_executor::IntentExecutor;
+use executor_core::native_task::{NativeTask, NativeTaskWrapper};
+use executor_primitives::AccountId;
+use jsonrpsee::RpcModule;
+use native_task_handler::NativeTaskOk;
+use parity_scale_codec::Decode;
+use serde::{Deserialize, Serialize};
+use tracing::{debug, error};
+
+#[derive(Debug, Deserialize)]
+pub struct RequestLoanParams {
+	pub chain_id: u64,
+	pub wallet_index: u32,
+	pub smart_wallet_address: String,
+	pub collateral_ticker: String,
+	pub collateral_size: String, // size (balance) of collateral in smart wallet spot
+	pub lending_ratio: u32, // percentage of total collateral sold as spot and returned as lending, e.g., 50 = 50%
+}
+
+#[derive(Serialize, Clone)]
+pub struct RequestLoanResponse {
+	pub spot_sell_cloid: String,
+	pub hedge_open_cloid: String,
+	pub usdc_received: String,
+	pub spot_sell_tx_hash: Option<String>,
+	pub hedge_open_tx_hash: Option<String>,
+}
+
+pub fn register_request_loan<
+	EthereumIntentExecutor: IntentExecutor + Send + Sync + 'static,
+	SolanaIntentExecutor: IntentExecutor + Send + Sync + 'static,
+	CrossChainIntentExecutor: IntentExecutor + Send + Sync + 'static,
+>(
+	module: &mut RpcModule<
+		RpcContext<EthereumIntentExecutor, SolanaIntentExecutor, CrossChainIntentExecutor>,
+	>,
+) {
+	module
+		.register_async_method("omni_requestLoan", |params, ctx, ext| async move {
+			let user = check_auth(&ext).map_err(|e| {
+				error!("Authentication check failed: {:?}", e);
+				PumpxRpcError::from(
+					DetailedError::new(AUTH_VERIFICATION_FAILED_CODE, "Authentication failed")
+						.with_suggestion("Please provide valid authentication credentials"),
+				)
+			})?;
+
+			let params = params.parse::<RequestLoanParams>().map_err(|e| {
+				error!("Failed to parse params: {:?}", e);
+				PumpxRpcError::from(
+					DetailedError::new(PARSE_ERROR_CODE, "Failed to parse request parameters")
+						.with_suggestion(format!("Invalid JSON structure: {}", e)),
+				)
+			})?;
+
+			debug!("Received omni_requestLoan, params: {:?}", params);
+
+			validate_chain_id(params.chain_id as u32, Some("evm")).map_err(PumpxRpcError::from)?;
+
+			validate_wallet_index(params.wallet_index).map_err(PumpxRpcError::from)?;
+
+			// Validate smart wallet address
+			let smart_wallet_address_bytes =
+				validate_omni_account_hex(&params.smart_wallet_address, "smart_wallet_address")
```

Comment:
Should this be `validate_ethereum_address` instead ? 
Then 20 len check is redundant